1xx-enable-ssl-for-kafka-bridge.md

```diff
@@ -0,0 +1,155 @@
+
+## Motivation
+
+The connection between HTTP Bridge and Kafka cluster can be secured. However, there is currently no support for securing client connections to the HTTP Bridge. The current recommendation is to use firewalls or API gateways to secure the client connections. This requires users to make additional configurations and does not provide the best user experience. 
+
+Moreover, some of the endpoints are for internal use only, such as `/healthy`, `/ready` and `/metrics` but they are exposed on the same listener and the same port as the other endpoints that are used by external clients. This means the endpoints for internal use only are also exposed to external clients. And if SSL is enabled, it will introduce more operational complexity as these internal endpoints would need to be configured with TLS and certificates.
+
+## Proposal
+
+This proposal introduces adds new configurations to enable an SSL server and load a keystore certificate and key in PEM format so that client connections to HTTP Bridge can be encrypted. The new configurations will be added to the existing `http` prefixed configurations:
+- http.ssl.enable
+- http.ssl.keystore.certificate.location
+- http.ssl.keystore.key.location
+- http.ssl.keystore.certificate.chain
+- http.ssl.keystore.key
+- http.ssl.enabled.protocols
+- http.ssl.enabled.cipher.suites
+- http.management.port
+
+When `http.ssl.enable` is set to `true`, the HTTP Bridge server starts with SSL enabled and the new configurations will allow users to define locations of the keystore files or certificate chain and key in `PEM` format. Additional formats such as `PKCS12` and `JKS` might be supported in the future if users request them.
+
+`http.ssl.enabled.protocols` can be configured with a comma separated list of enabled secure transport protocols that the server will accept from connecting clients. If not set by the user, the server will use `TLSv1.2,TLSv1.3` as the default. 
+
+`http.ssl.enabled.cipher.suites` can be configured with a comma separated list of cipher suites that the server will support. If not set, the default list of cipher suites provided by the underlying JDK SSL/TLS engine will be used.
+
+If the existing `http.port` is not set and `http.ssl.enable` is set to `true`, the server port will be set to `443` by default. This will allow users to make TLS encrypted connections from their clients for all the HTTP Bridge endpoints for client operations.
+
+This proposal also adds support for a separate listener for internal endpoints such as:
+- `/healthy`
+- `/ready`
+- `/metrics`
+
+The internal endpoints will use the port defined by the new configuration property `http.management.port`, defaulting to 8081 if unspecified. These endpoints will no longer be accessible via the default listener port defined by `http.port` or port 8080. All other endpoints will continue to use the existing listener on port 8080, or 443 if SSL is enabled.
```

Comment:
Did you try what happens if ... the user DOESN'T enable TLS so they get default HTTP port 8080 but then they also set `http.management.port` to 8080? If you start two servers in the bridge, it's not going to work because the first one will keep the port and the other one will find the port already used. At the same time, as user, I would expect this to work. I just want operations and admin on the same 8080 port as today. Have you taken into account this scenario in terms of implementation?